webview_panels/src/modules/queryPanel/components/perspective/PerspectivePlugins.js

```diff
@@ -0,0 +1,174 @@
+    td.textContent = "";
+
+    // Adding svg icon to the span element
+    const span = document.createElement("span");
+    const img = document.createElement("img");
+
+    // Getting the current theme and setting the icon color
+    const currentTheme = getGlobalTheme();
+    const iconColor = currentTheme === "dark" ? "white" : "black";
+    img.src = `data:image/svg+xml,%3Csvg%20xmlns='http://www.w3.org/2000/svg'%20viewBox='0%200%2030%2030'%20width='12px'%20fill='${iconColor}'%20height='12px'%3E%3Cpath%20d='M%2025.980469%202.9902344%20A%201.0001%201.0001%200%200%200%2025.869141%203%20L%2020%203%20A%201.0001%201.0001%200%201%200%2020%205%20L%2023.585938%205%20L%2013.292969%2015.292969%20A%201.0001%201.0001%200%201%200%2014.707031%2016.707031%20L%2025%206.4140625%20L%2025%2010%20A%201.0001%201.0001%200%201%200%2027%2010%20L%2027%204.1269531%20A%201.0001%201.0001%200%200%200%2025.980469%202.9902344%20z%20M%206%207%20C%204.9069372%207%204%207.9069372%204%209%20L%204%2024%20C%204%2025.093063%204.9069372%2026%206%2026%20L%2021%2026%20C%2022.093063%2026%2023%2025.093063%2023%2024%20L%2023%2014%20L%2023%2011.421875%20L%2021%2013.421875%20L%2021%2016%20L%2021%2024%20L%206%2024%20L%206%209%20L%2014%209%20L%2016%209%20L%2016.578125%209%20L%2018.578125%207%20L%2016%207%20L%2014%207%20L%206%207%20z'%3E%3C/path%3E%3C/svg%3E`;
+
+    // Styling the image element
+    img.style.paddingTop = "5px";
+
+    // Styling the span element
+    span.style.marginLeft = "5px";
+    span.style.color = "var(--primary-color)";
+
+    // Styling the div element
+    div.style.display = "flex";
+    div.style.alignItems = "center";
+    div.style.justifyContent = "space-between";
+
+    // Styling the text element
+    text.style.overflow = "hidden";
+    text.style.textOverflow = "ellipsis";
+    text.style.whiteSpace = "nowrap";
+    text.style.width = "100px";
+
+    // Appending the image to the span element and the text to the div element
+    // Appending the div element to the td element
+    span.appendChild(img);
+    div.appendChild(text);
+    div.appendChild(span);
+    td.appendChild(div);
+  }
+}
+
+// Adds click event to the td element
+function makeClickEvent(td, metadata) {
+  // If string length is greater than 20, truncate and add ellipsis
+  if (metadata.value.length > 20) {
+    // td.textContent = metadata.value.slice(0, 20) + "...";
+    td.style.cursor = "pointer";
+  }
+
+  const columnName = metadata.column_header;
+  if (isJson(metadata.value)) {
+    appendImage(td);
+    td.style.cursor = "pointer";
+    // If string is a valid JSON, add click event to view JSON
+    td.onclick = () => {
+      dispatchCustomEvent(
+        "string-json-viewer",
+        metadata.value,
+        columnName,
+        "json"
+      );
+    };
+  } else if (metadata.value.length > 20) {
+    appendImage(td);
+    // else if string is greater than 20, add click event to view full string
+    td.onclick = () => {
+      dispatchCustomEvent(
```

Comment:
it would be more useful if we emit this event as `onCellClick` in perspectiveviewer props, instead of dispatching to window. We can make use the click event for any functionality in future (for ex: copying the cell content). Keeping it in props will be more maintainable

Reply:
We aren't using that component in the js, so its props don't work there. I'm using events to communicate the drawer opening.